### PR TITLE
Display the current editor tool at the mouse cursor

### LINKED ISSUE #491: Editor: Show currently selected tool at mouse position
I often find myself confused which tool is currently selected and whether I am in tile select or object select mode.

I think it would be good if we could show the currently selected tool near the mouse pointer.

### PR DESCRIPTION
**Pull request by Hume2**
fixes #491

### CODE REVIEW

File: src/editor/tool_icon.cpp [hunk 1]
```diff
@@ -56,4 +56,9 @@ ToolIcon::next_mode() {
   }
 }
 
+SurfacePtr
```
Comment (tobbi) [inline src/editor/tool_icon.cpp]: const

Comment (Hume2) [inline src/editor/tool_icon.cpp]: Done, it's on the line just bellow.

**Comment by maxteufel** [comment]
In #491, @tobbi said he also wanted to see if he's in object select mode, this pull request doesn't really implement that part.

**Comment by maxteufel** [comment]
Looks good to merge once the builds succeed. :+1:

File: src/gui/mousecursor.hpp [hunk 2]
```diff
@@ -58,6 +58,9 @@ class MouseCursor
   /// Draw MouseCursor on screen.
   void draw(DrawingContext& context);
 
+  /// Set the icon
+  void set_icon(SurfacePtr icon_);
```
Comment (tobbi) [inline src/gui/mousecursor.hpp]: const SurfacePtr& icon_

File: src/gui/mousecursor.cpp [hunk 3]
```diff
@@ -53,6 +54,11 @@ void MouseCursor::set_mid(int x, int y)
   m_mid_y = y;
 }
 
+void MouseCursor::set_icon(SurfacePtr icon_)
```
Comment (tobbi) [inline src/gui/mousecursor.cpp]: Same here:
const SurfacePtr& icon_

Comment (Hume2) [inline src/gui/mousecursor.cpp]: This doesn't work, because set_icon(nullptr). SurfacePtr is std::shared_ptr<Surface>, so it is already a pointer.

**Comment by tobbi** [comment]
I think a better position for this is in the following screenshot:
http://screencast.com/t/kQPVBVluf4Zm (red square).

What does the rest of the team think? IMO it's more natural that way.

**Comment by Rusty-Box** [comment]
@tobbi: I think it looks good.

**Pull request merged**